src/mount-support-nvidia.c

```diff
@@ -187,39 +189,56 @@ static void sc_mount_nvidia_driver_arch(const char *rootfs_dir)
 #endif				// ifdef NVIDIA_ARCH
 
 #ifdef NVIDIA_UBUNTU
+
+struct sc_nvidia_driver {
+	int major_version;
+	int minor_version;
+};
+
+#define SC_NVIDIA_DRIVER_VERSION_FILE "/sys/module/nvidia/version"
+#define SC_LIBGL_DIR "/var/lib/snapd/lib/gl"
+
+static void sc_probe_nvidia_driver(struct sc_nvidia_driver *driver)
+{
+	FILE *file __attribute__ ((cleanup(sc_cleanup_file))) = NULL;
+	debug("opening file describing nvidia driver version");
+	file = fopen(SC_NVIDIA_DRIVER_VERSION_FILE, "rt");
```

Comment:
Yeah, this begs for some more testability. I'm working on improvements but I'd like to land it so that we can release it quickly as well.